packages/components/src/components/common/inputtag/inputBadge.tsx

```diff
@@ -0,0 +1,79 @@
+import { forwardRef, type ElementRef } from 'react';
+import { useField } from '../field/field';
+import { IvyIcon } from '../icon/icon';
+import { inputBadge, inputBadgeIcon, inputBadgeLine, inputBadgeOutput } from './inputBadge.css';
+
+import type { IvyIcons } from '@axonivy/ui-icons';
+
+export type InputBadgeProps = {
+  value: string;
+  badgeRegex: BadgeRegex[];
+};
+
+type BadgeRegex = { delimiter: { start: string; end: string }; icon: IvyIcons; badgeTextGen: (text: string) => string };
+
+export type BadgeTextProps = { icon: IvyIcons | undefined; text: string };
+
+const Badge = ({ badge }: { badge: BadgeTextProps }) => {
+  if (badge.icon) {
+    return (
+      <span className={inputBadge}>
+        <span className={inputBadgeIcon}>
+          <IvyIcon icon={badge.icon} />
+        </span>
+        {badge.text}
+      </span>
+    );
+  }
+  return <span>{badge.text}</span>;
+};
+
+export const InputBadge = ({ value, badgeRegex, ...props }: InputBadgeProps) => {
+  const { inputProps } = useField();
+  const badges = findBadges(value, badgeRegex);
+  return (
+    <output className={inputBadgeOutput} {...inputProps} {...props}>
+      {badges.map((badge, index) => (
+        <Badge key={index} badge={badge} />
+      ))}
+    </output>
+  );
+};
+
+export const InputBadgeArea = forwardRef<ElementRef<'output'>, InputBadgeProps>(({ value, badgeRegex, ...props }, forwardRef) => {
+  const { inputProps } = useField();
+  const lines = value.split(/\r\n|\r|\n/).map(e => findBadges(e, badgeRegex));
+  return (
+    <output className={inputBadgeOutput} {...inputProps} {...props} ref={forwardRef}>
+      {lines.map((line, index) => (
+        <div key={index} className={inputBadgeLine} role='row'>
+          {line.map((badge, index) => (
+            <Badge key={index} badge={badge} />
+          ))}
+        </div>
+      ))}
+    </output>
+  );
+});
+
+const findBadges = (value: string, badgeRegex: BadgeRegex[]): BadgeTextProps[] => {
+  const regexStrings: string[] = [];
+  badgeRegex.forEach(reg => {
+    regexStrings.push('(?:' + reg.delimiter.start + '.*?' + reg.delimiter.end + ')');
+  });
+  const separated = value.split(new RegExp('(' + regexStrings.join('|') + ')'));
```

Comment:
we could do this like that, but I can imagine that this is not ideal if we do this everytime. Why shouldn't we just require a regex in the BadgeRegex type? User of this component are devs, I think they should be able to create such a regex, especially if they have examples.